pkg/controller/hyperconverged/hyperconverged_controller_test.go

```diff
@@ -279,6 +283,34 @@ var _ = Describe("HyperconvergedController", func() {
 				}
 			})
 
+			It("should deploy templates to different namespace", func() {
+				ci := hcoutil.GetClusterInfo()
+				if !ci.IsOpenshift() {
+					Skip("skipping due to missing template kind")
+				}
+
+				expected := getBasicDeployment()
+				expected.hco.Spec.CommonTemplatesNamespace = &expected.hco.Namespace
+
+				cl := expected.initClient()
+				r := initReconciler(cl, nil)
+
+				// Do the reconcile
+				res, err := r.Reconcile(context.TODO(), request)
+				Expect(err).To(BeNil())
+				Expect(res).Should(Equal(reconcile.Result{}))
+
+				templatesList := &templatev1.TemplateList{}
+				Eventually(func() bool {
+					err := cl.List(context.TODO(),
+						templatesList,
+						client.InNamespace(namespace),
+						client.MatchingLabels{
+							templateTypeLabel: base,
+						})
+					return len(templatesList.Items) > 0 && err == nil
+				}, 30*time.Second).Should(BeTrue(), "templates should be deployed in different namespace")
+			})
```

Comment:
This test is a unit test, without a running cluster. There is no running SSP to implement the logic under test. Also, as this is a unit test, it should not test other components. Please just make sure the required fields are set in the SSP CR.

Reply:
ah I see, I updated the test